Userland/Libraries/LibTLS/Socket.cpp

```diff
@@ -57,99 +61,112 @@ String TLSv12::read_line(size_t max_size)
     return line;
 }
 
-bool TLSv12::write(ReadonlyBytes buffer)
+ErrorOr<size_t> TLSv12::write(ReadonlyBytes bytes)
 {
     if (m_context.connection_status != ConnectionStatus::Established) {
         dbgln_if(TLS_DEBUG, "write request while not connected");
-        return false;
+        return AK::Error::from_string_literal("TLS write request while not connected");
     }
 
-    for (size_t offset = 0; offset < buffer.size(); offset += MaximumApplicationDataChunkSize) {
-        PacketBuilder builder { MessageType::ApplicationData, m_context.options.version, buffer.size() - offset };
-        builder.append(buffer.slice(offset, min(buffer.size() - offset, MaximumApplicationDataChunkSize)));
+    for (size_t offset = 0; offset < bytes.size(); offset += MaximumApplicationDataChunkSize) {
+        PacketBuilder builder { MessageType::ApplicationData, m_context.options.version, bytes.size() - offset };
+        builder.append(bytes.slice(offset, min(bytes.size() - offset, MaximumApplicationDataChunkSize)));
         auto packet = builder.build();
 
         update_packet(packet);
         write_packet(packet);
     }
 
-    return true;
+    return bytes.size();
 }
 
-bool TLSv12::connect(const String& hostname, int port)
+ErrorOr<NonnullOwnPtr<TLSv12>> TLSv12::connect(const String& host, u16 port, Options options)
 {
-    set_sni(hostname);
-    return Core::Socket::connect(hostname, port);
+    Core::EventLoop loop;
```

Comment:
Uh, this is very weird. Why are we creating a miniature `EventLoop` in this function?

Reply:
To block until we've either successfully connected, or failed somehow (`TLS::connect()` shouldn't give you a socket that's half-connected)